drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -3664,6 +3691,7 @@ static struct rswitch_ipv4_route *rswitch_get_route(struct rswitch_private *priv
 	bool default_present = false;
 
 	mutex_lock(&priv->rdev_list_lock);
+	mutex_lock(&priv->ipv4_forward_lock);
```

Comment:
It seems like this lock can be redundant because forward flag status has to be kept in mind only when you setup hw.